### PR TITLE
348840043: Migrated Safari cookies construct-based parser to use dtfabric #1893

### PR DESCRIPTION
**Pull request by joachimmetz**
[Code review: 348840043: Migrated Safari cookies construct-based parser to use dtfabric #1893](https://codereview.appspot.com/348840043/)

### CODE REVIEW

File: plaso/parsers/safari_cookies.py [hunk 1]
```diff
@@ -43,104 +43,127 @@ def __init__(self):
...
-  _PAGE_HEADER = construct.Struct(
-      'page_header',
-      construct.Bytes('header', 4),
-      construct.ULInt32('number_of_records'),
-      construct.Array(
-          lambda ctx: ctx.number_of_records,
-          construct.ULInt32('offsets')))
+  _DEFINITION_FILE = 'safari_cookies.yaml'
+
+  _SIGNATURE = b'cook'
```
Comment (Onager) [inline plaso/parsers/safari_cookies.py]: Add the null byte here as well, to match the format specification

File: plaso/parsers/safari_cookies.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  number_of_elements: 4
+- name: number_of_pages
+  data_type: uint32
+---
+name: binarycookies_page_sizes
+type: sequence
+element_data_type: uint32be
+number_of_elements: binarycookies_file_header.number_of_pages
+---
+name: binarycookies_file_footer
```
Comment (Onager) [inline plaso/parsers/safari_cookies.yaml]: Please remove the footer, as it's not used

File: plaso/parsers/safari_cookies.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/parsers/safari_cookies.py]: no the signature is 4 bytes the 0-byte in the format spec is there to reduce false positives

File: plaso/parsers/safari_cookies.yaml (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/safari_cookies.yaml]: ack

**Pull request closed without merging**